### DISCUSSION TITLE
Alternate open-source solver

### DISCUSSION THREAD

**Discussion by secretlyagoblin** [cdriesler/glasshopper.io#157] [Ideas & Requests]
The logical end point of this project (seems to me) should be tearing out the Compute engine entirely and building an open source solver that can be built out.

Start with adding and subtraction, work up to meshes, and then... who knows, the seed of something is there.

Don't fix what aint broke - NodePen is a love letter to grasshopper, let grasshopper be the template.

I don't have the time or technical chops to think about actually architecting a DAG solver, but I would readily contribute to a framework if one was in place.

**Comment by cdriesler** [comment]
You're not wrong! The new API is definitely designed to be this agnostic about what it's executing on.

But, say we did manage to create a kickass open source solver. At that point, the question for me is: why use NodePen tech for the frontend? There are so many DAG visualization & interaction libraries out there. Constantly asking myself: If NodePen is to be a DAG library for DAG's sake, what does it offer that's any better than, say, [react-flow](https://reactflow.dev/)?

I don't have a good answer right now, but I'm also not yet taking the question too seriously. It's a logical line of questioning that comes naturally out of making the scope more generic. But I still have a lot to deliver on the promise that it can do Grasshopper.

Maybe it's not about being generically _better_ than other libraries, but _best_ for a narrower set of concerns? Fun to think about, regardless.

**Reply by secretlyagoblin** [reply]
I raised this at work and one of our lead developers said the exact same thing, so I think your instinct is correct!
